### PR TITLE
[googletest] bump from v1.10 to v1.13 to fix compile issues

### PR DESCRIPTION
**Pull request by tmm1**
fixes compile issues on my system:

```
inputstream.adaptive/build/build/googletest/src/googletest/googlemock/include/gmock/gmock-matchers.h:891:3: error: definition of implicit copy constructor for 'EndsWithMatcher<std::wstring>' is deprecated because it has a user-declared copy assignment operator [-Werror,-Wdeprecated-copy]
  GTEST_DISALLOW_ASSIGN_(EndsWithMatcher);
  ^
```

which is a known issue fixed here: https://github.com/google/googletest/pull/2815

my compiler details:

```
Apple clang version 14.0.0 (clang-1400.0.29.202)
Target: arm64-apple-darwin22.2.0
Thread model: posix
InstalledDir: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin
```

### CODE REVIEW

File: depends/common/googletest/googletest.txt [hunk 1]
```diff
@@ -1 +1 @@
-googletest http://mirrors.kodi.tv/build-deps/sources/googletest-1.10.0.tar.gz
+googletest https://github.com/google/googletest/archive/refs/tags/v1.13.0.tar.gz
```
Comment (fuzzard) [inline depends/common/googletest/googletest.txt]: Ive thrown this up on mirrors. Should start being available shortly

https://mirrors.kodi.tv/build-deps/sources/googletest-1.13.0.tar.gz

**Comment by CastagnaIT** [comment]
i think this version bring some problem when compiling on windows UWP systems
this is a log example:
https://dev.azure.com/teamkodi/c0b097aa-5a38-4262-8355-1f4897fea12d/_apis/build/builds/12342/logs/68
but i am no expert on builds
ofc need to be fixed

**Comment by CastagnaIT** [comment]
is possible we need to add new `GTEST_HAS_FILE_SYSTEM` flag?
see: https://github.com/google/googletest/commit/b0846aaf37a271d56019b42121ebcea0ac42e8c1

**Comment by CastagnaIT** [comment]
Are you planning to investigate to finish the PR in the near future?
just to know if we have to take it over,
in which case i close it and i will add it to the roadmap to be done when we have time

**Comment by CastagnaIT** [comment]
no answers and since cannot be merged as is
i close the PR and i add it to roadmap
if one day you want continue the development we can always reopen it

**Pull request closed without merging**